### DISCUSSION TITLE
EventHandler + Condition (false) => What happens to the message? And RabbitMQ stability

### DISCUSSION THREAD

**Discussion by rmindel** [Netflix/conductor#2325] [Q&A]
I have a queue with work items.
I want to have 2 different actions depending on a property within the work item message.

If I create 2 event handlers on the same queue, with 2 conditions:
EH1.Condition - $.Data == 1
EH2.Condition - $.Data == 2

If EH1 takes a message with Data = 2, will the EH return that message to the queue, or pass it to the other EH somehow?
Or do I need 2 queues for 2 Event Handlers?

Also dealing with event handlers with rabbit mq queues seems a little unstable.
When disabling an EventHandler that's consuming a rabbitMQ queue, the consumer stays on the queue and messages keep getting removed from that queue.

so I was wondering what's the correct way of dealing with EH (when creating, removing, disabling and updating them)

Thanks!!

**Comment by apanicker-nflx** [comment]
>will the EH return that message to the queue, or pass it to the other EH somehow?
Or do I need 2 queues for 2 Event Handlers?

Once a message is dequeued from an event queue, it will be processed by all event handlers that are configured for that queue. The actions on the event handler will only be executed for the EH for which the condition is satisfied. For the EH that do not satisfy the condition, the actions will be skipped.

@ritu-p Could you please help with the event handler issue that @rmindel is running into with RabbitMQ queues?

**Comment by ritu-p** [comment]
Hi @rmindel ,
We for our requirement too had used 2 seperate evet handlers with 2 diffrent conditions to execute and action.However as you corrcetly said both are created on the same queue.

By disabling do you mean deleting an event handler? I think deleting handles diconnection.I have to look into the code to tell for sure.
There are some other enhancements needed like retryingin case a rabbitmq connectivity fails which I need to fix in the code. Please fo raise a bug with the issue if possible as I am not actively working on this code. If I can I will try to fix it.
